signal2html/versioninfo.py

```diff
@@ -0,0 +1,33 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+"""Class grouping version-specific information from the signal database.
+
+License: See LICENSE file.
+"""
+
+
+class VersionInfo(object):
+    def __init__(self, version):
+        self.version = int(version)
+
+    def is_tested_version(self) -> bool:
+        """Returns whether the database version has been tested.
+
+        Testing and pull requests welcome."""
+
+        return self.version in [18, 23, 65, 80, 89]
+
+    def is_addressbook_using_rids(self) -> bool:
+        """Returns whether the contacts are structured using recipient IDs.
+
+        Previous versions referred to contacts using their phone numbers or a
+        special group ID. Current knowledge: change happened strictly after
+        version 23 and at the latest at version 65."""
+
+        return self.version > 23
+
+    def get_reactions_query_column(self) -> str:
+        """Returns a SQL expression to retrieve reactions to MMS messages."""
+
+        return "reactions" if self.version >= 89 else "''"
```

Comment:
Hi @GjjvdBurg could you check if that column is present in any of your prior backups (65 / 80)?

Reply:
I don't have a backup with that version, but if I've understood correctly, [this line/file](https://github.com/signalapp/Signal-Android/blob/64ebf20c1b271f3f02a5d8023136d8fa8086d5b5/app/src/main/java/org/thoughtcrime/securesms/database/helpers/SQLCipherOpenHelper.java#L712) shows that reactions were introduced in version 37.

Reply:
Ahah, thanks for the pointer!